framework/src/modules/interoperability/base_interoperability_endpoint.ts

```diff
@@ -0,0 +1,166 @@
+	}
+
+	public async getChainAccount(
+		context: ModuleEndpointContext,
+		chainID: Buffer,
+	): Promise<ChainAccountJSON> {
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+		return this._getDesiredChainAccount(await interoperabilityStore.getChainAccount(chainID));
+	}
+
+	public async getAllChainAccounts(
+		context: ModuleEndpointContext,
+		startChainID: Buffer,
+	): Promise<{ chains: ChainAccountJSON[] }> {
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+
+		const chainAccounts = (
+			await interoperabilityStore.getAllChainAccounts(startChainID)
+		).map(chainAccount => this._getDesiredChainAccount(chainAccount));
+
+		return { chains: chainAccounts };
+	}
+
+	public async getChannel(
+		context: ModuleEndpointContext,
+		chainID: Buffer,
+	): Promise<ChannelDataJSON> {
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+
+		const {
+			inbox,
+			messageFeeTokenID,
+			outbox,
+			partnerChainOutboxRoot,
+		} = await interoperabilityStore.getChannel(chainID);
+
+		const inboxJSON = this._toBoxJSON(inbox) as InboxJSON;
+		const outboxJSON = this._toBoxJSON(outbox) as OutboxJSON;
+
+		const messageFeeTokenIDJSON: MessageFeeTokenIDJSON = {
+			chainID: messageFeeTokenID.chainID.toString('hex'),
+			localID: messageFeeTokenID.localID.toString('hex'),
+		};
+
+		return {
+			messageFeeTokenID: messageFeeTokenIDJSON,
+			outbox: outboxJSON,
+			inbox: inboxJSON,
+			partnerChainOutboxRoot: partnerChainOutboxRoot.toString('hex'),
+		};
+	}
+
+	public async getOwnChainAccount(context: ModuleEndpointContext): Promise<OwnChainAccountJSON> {
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+
+		const { id, name, nonce } = await interoperabilityStore.getOwnChainAccount();
+
+		return {
+			id: id.toString('hex'),
+			name,
+			nonce: nonce.toString(),
+		};
+	}
+
+	public async getTerminatedStateAccount(
+		context: ModuleEndpointContext,
+		chainID: Buffer,
+	): Promise<TerminatedStateAccountJSON> {
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+
+		const {
+			stateRoot,
+			initialized,
+			mainchainStateRoot,
+		} = await interoperabilityStore.getTerminatedStateAccount(chainID);
+
+		return {
+			stateRoot: stateRoot.toString('hex'),
+			initialized,
+			mainchainStateRoot: mainchainStateRoot?.toString('hex'),
+		};
+	}
+
+	public async getTerminatedOutboxAccount(
+		context: ModuleEndpointContext,
+		chainID: Buffer,
+	): Promise<TerminatedOutboxAccountJSON> {
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+
+		const {
+			outboxRoot,
+			outboxSize,
+			partnerChainInboxSize,
+		} = await interoperabilityStore.getTerminatedOutboxAccount(chainID);
+
+		return {
+			outboxRoot: outboxRoot.toString('hex'),
+			outboxSize,
+			partnerChainInboxSize,
+		};
+	}
+
+	private _getDesiredChainAccount(chainAccount: ChainAccount) {
```

Comment:
this could be a util function as it does not use any class members so we could extract it to an independent function in the same file and export, also better to rename it to `chainAccountToJSON` similar to other util functions